core/station/impl/src/services/external_canister.rs

```diff
@@ -718,32 +770,176 @@ impl ExternalCanisterService {
+        if let Some(permission) = input.change {
+            self.permission_service
+                .edit_permission(EditPermissionOperationInput {
+                    auth_scope: Some(permission.auth_scope),
+                    users: Some(permission.users),
+                    user_groups: Some(permission.user_groups),
+                    resource: Resource::ExternalCanister(ExternalCanisterResourceAction::Change(
+                        ExternalCanisterId::Canister(external_canister.canister_id),
+                    )),
+                })?;
         }
 
-        if let Some(updated_request_policies) = input.request_policies {
-            self.configure_external_canister_request_policies(
-                &external_canister,
-                updated_request_policies,
-            )?;
+        // calls permissions (for calling methods of the external canister)
+        if let Some(calls_permissions) = input.calls {
+            match calls_permissions {
+                ExternalCanisterChangeCallPermissionsInput::ReplaceAllBy(calls) => {
+                    // removes all existing call permissions
+                    self.permission_repository
+                        .find_external_canister_call_permissions(&external_canister.canister_id)
+                        .iter()
+                        .for_each(|permission| {
+                            self.permission_service
+                                .remove_permission(&permission.resource);
+                        });
+
+                    // adds the new call permissions
+                    for call in calls {
+                        let call_resource =
+                            Resource::ExternalCanister(ExternalCanisterResourceAction::Call(
+                                CallExternalCanisterResourceTarget {
+                                    execution_method:
+                                        ExecutionMethodResourceTarget::ExecutionMethod(
+                                            CanisterMethod {
+                                                canister_id: external_canister.canister_id,
+                                                method_name: call.execution_method,
+                                            },
+                                        ),
+                                    validation_method: call.validation_method,
+                                },
+                            ));
+
+                        self.permission_service
+                            .edit_permission(EditPermissionOperationInput {
+                                auth_scope: Some(call.allow.auth_scope),
+                                users: Some(call.allow.users),
+                                user_groups: Some(call.allow.user_groups),
+                                resource: call_resource.clone(),
+                            })?;
+                    }
+                }
```

Comment:
could we use a helper similarly to `maybe_mutate_canister_calls_request_policies`?

Reply:
just added `maybe_mutate_canister_calls_permissions`